src/main/java/smartthings/konsumer/filterchain/MessageFilterChain.java

```diff
@@ -0,0 +1,57 @@
+package smartthings.konsumer.filterchain;
+
+import kafka.message.MessageAndMetadata;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import smartthings.konsumer.MessageProcessor;
+import smartthings.konsumer.circuitbreaker.CircuitBreaker;
+
+import java.util.*;
+
+public class MessageFilterChain {
```

Comment:
Yeah, the MessageFilterChain was meant as a wrapper around the filters responsible for making sure they get called in the correct order, even when a filter calls next more than once.